src/elasticarray.jl

```diff
@@ -57,76 +67,139 @@ ElasticArray(A::AbstractArray{T,N}) where {T,N} = copyto!(ElasticArray{T,N}(unde
 Base.convert(::Type{T}, A::AbstractArray) where {T<:ElasticArray} = A isa T ? A : T(A)
 
 
+@inline function Base.similar(A::ElasticArray, ::Type{T}, dims::Dims{N}) where {T,N}
+    kernel_size, size_lastdim = _split_dims(dims)
+    kernel_length = prod(kernel_size)
+    data = similar(A.data, T, prod(dims))
+    ElasticArray{T,N}(kernel_size, SignedMultiplicativeInverse{Int}(kernel_length), data)
+end
+
+@inline function Base.:(==)(A::ElasticArray, B::ElasticArray)
+    return ndims(A) == ndims(B) && A.kernel_size == B.kernel_size && A.data == B.data
+end
+
+
+@inline Base.size(A::ElasticArray) = (A.kernel_size..., div(length(eachindex(A.data)), A.kernel_length))
 
+@inline Base.length(A::ElasticArray) = length(A.data)
 
-import Base.==
-(==)(A::ElasticArray, B::ElasticArray) =
-    ndims(A) == ndims(B) && A.kernel_size == B.kernel_size && A.data == B.data
+@propagate_inbounds Base.getindex(A::ElasticArray, i::Int) = getindex(A.data, i)
 
+@propagate_inbounds Base.setindex!(A::ElasticArray, x, i::Int) = setindex!(A.data, x, i)
 
-Base.parent(A::ElasticArray) = A.data
```

Comment:
Because at least one stdlib function depends on `size(A::AbstractArray) == size(parent(similar(A, ...)))`. Here's a snippet from `LinearAlgebra.inv`:
```julia
Ai = inv!(lu(AA))
Ai = convert(typeof(parent(Ai)), Ai)
```

Before adding `Base.similar(A::ElasticArray, ...)`, `inv!(lu(AA))` returned an `Matrix`, but now it returns an `ElasticArray`. So when we do `convert(typeof(parent(Ai)), Ai)` we try to convert a matrix into a vector (because `typeof(parent(Ai)) <: AbstractVector`), which fails.

If we change `Base.similar` to `Base.similar(A::ElasticArray, T::Type, dims::Dims) = similar(A.data, T, dims)` as mentioned above then we can add back in `Base.parent`.

But from the docs for `Base.parent`
> Returns the "parent array" of an array view type (e.g., SubArray), or the array
  itself if it is not a view.

Is `ElasticArray` really a "view type"? Do we need `Base.parent`? I'm unsure :)